content/docs/reference-javascript-environment-requirements.md

```diff
@@ -6,9 +6,11 @@ category: Reference
 permalink: docs/javascript-environment-requirements.html
 ---
 
-React 16 depends on the collection types [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) and [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set). If you support older browsers and devices which may not yet provide these natively (e.g. IE < 11) or which have non-compliant implementations (e.g. IE 11), consider including a global polyfill in your bundled application, such as [core-js](https://github.com/zloirock/core-js).
+React 18 depends on modern browser features including [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) and [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set).
 
-A polyfilled environment for React 16 using core-js to support older browsers might look like:
+If you support older browsers and devices which do not provide a native microtask-based Promise implementation such as Internet Explorer, we recommend you stay with React 17, because microtasks cannot be polyfilled without impacting performance in React 18.  
+
+To support older browsers which do not include `Map` or `Set`, consider using [core-js](https://github.com/zloirock/core-js):
```

Comment:
Do we actually recommend core-js for anything anymore?

`Map` and `Set` were already in IE11, at least for how we used them.

So I think it's only `Symbol` and `Object.assign` that we now require polyfilling. Is `core-js` what we'd recommend for that?

Reply:
Updated! I think the best practice now is to configure Browserlist if available, otherwise provide the polyfill directly. So I removed the core-js example and explained the current state.